### ISSUE TITLE
Add doctests for examples

### ISSUE THREAD

**Issue by energizah** [NixOS/nixpkgs#86612]
Sometimes the documentation examples have errors. For instance, in [`hardware.printers.ensurePrinters.*.model`](https://github.com/NixOS/nixpkgs/blob/0107ee8c322e57cdb2ebcc1c9c4286ff7db53d5c/nixos/modules/hardware/printers.nix#L87), an example refers to `lib.version.majorMinor`, but this gives an error because `lib.version` is a string (the correct attribute is `lib.versions`).

[Doctests](https://en.wikipedia.org/wiki/Doctest) are a quality-control feature that attempts to evaluate example code in documentation, to verify that the documentation is correct. Running doctests on examples can catch errors like this one, and can also ensure that examples stay up-to-date.

**Comment by FRidh** [comment]
I agree with you. Here we need to be careful though because we do not want to build those expressions.
